compiler/noirc_frontend/src/ast/expression.rs

```diff
@@ -71,6 +72,21 @@ impl UnresolvedGeneric {
             }
         }
     }
+
+    pub fn kind(&self) -> Kind {
+        match self {
+            UnresolvedGeneric::Variable(_) => Kind::Normal,
+            // The inner numeric type of the kind cannot be resolved from an `UnresolvedGeneric`
+            // on its own. It is expected that the inner type is updated during elaboration.
+            UnresolvedGeneric::Numeric { .. } => Kind::Numeric { typ: Box::new(Type::Error) },
```

Comment:
Also, really small nitpick (sorry): Since `Kind::Numeric` only has one field, and the `typ` name doesn't add particularly much, I think it'd be more terse if it were a tuple struct instead.